### PR TITLE
steal-effect: Add steal effect

### PR DESCRIPTION
**Pull request by asternotus**
1. Add steal_effect.py
2. Add utils.py
3. Change user_interaction.py realization
4. Change Effect \_\_init\_\_ method params

### CODE REVIEW

File: effects/steal_effect.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        while(True):
+            target_number = user_interaction.read_index("config.type_target")
+            if self.validate(target_number):
+                self.target = self.game.citizens[target_number - 1]
+                break
+
+    def resolve(self):
+        if self.targets[0].citizen_card is not None:
+
+            if isinstance(self.targets[0], Spy):
```
Comment (elias-gvin) [inline effects/steal_effect.py]: This logic should be performed inside SpyEffect, not in StealEffect

File: effects/steal_effect.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                self.creator.stolen_cards.append(self.targets[0].citizen_card)
+                self.targets[0].citizen_card = None
+                user_interaction.save_active("config.card_stolen")
+                user_interaction.save_passive("config.lost_card")
+
+        else:
+            user_interaction.save_active("config.no_card")
+
+    def validate(self, target_number):
+        return (utils.is_citizen_in_range(target_number, self.game.citizens) and self.game.citizen[target_number].is_alive)
```
Comment (elias-gvin) [inline effects/steal_effect.py]: I'd rather create appropriate method in utils with same logic (and call this method from effect's validate.

Methid in utils will.look something like
def validate_citizen_target_number(...):
    return (target_number is not None
                 and is_citizen_in_range(target_number, citizens)
                 and citizen[target_number].is_alive)

File: effects/steal_effect.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+class StealEffect(Effect):
+
+    def __init__(self, name: str, game: Game, creator: Citizen) -> None:
+        Effect.__init__(self, name, game, creator)
+
+    def activate(self):
+        while(True):
+            target_number = user_interaction.read_index("config.type_target")
+            if self.validate(target_number):
+                self.target = self.game.citizens[target_number - 1]
```
Comment (elias-gvin) [inline effects/steal_effect.py]: Add message for user about input error in case id validate returns 'false'

File: effects/steal_effect.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import user_interaction
+import utils
+
+class StealEffect(Effect):
+
+    def __init__(self, name: str, game: Game, creator: Citizen) -> None:
+        Effect.__init__(self, name, game, creator)
+
+    def activate(self):
+        while(True):
```
Comment (elias-gvin) [inline effects/steal_effect.py]: Infinite lops aren't a good programming practice in most of the cases (except global game cycles, work with network and threads).
I'd suggest to add condition in this loop. But it is not necessary

**Review by elias-gvin — CHANGES_REQUESTED** [review]

File: effects/steal_effect.py (hunk 1 above)
Comment (asternotus) [reply effects/steal_effect.py]: Done

File: effects/steal_effect.py (hunk 2 above)
Comment (asternotus) [reply effects/steal_effect.py]: Done

File: effects/steal_effect.py (hunk 3 above)
Comment (asternotus) [reply effects/steal_effect.py]: Done

File: effects/steal_effect.py (hunk 4 above)
Comment (asternotus) [reply effects/steal_effect.py]: Done

**Review by elias-gvin — APPROVED** [review]

**Pull request merged**